.circleci/config.yml

```diff
@@ -0,0 +1,72 @@
+version: 2.1
+
+references:
+  images:
+    go: &GOLANG_IMAGE circleci/golang:latest
+
+# reusable 'executor' object for jobs
+executors:
+  go:
+    docker:
+      - image: *GOLANG_IMAGE
+    environment:
+      - TEST_RESULTS: /tmp/test-results # path to where test results are saved
+
+jobs:
+  go-fmt-and-vet:
+    executor: go
+    steps:
+      - checkout
+
+      # Restore go module cache if there is one
+      - restore_cache:
+          keys:
+            - go-version-modcache-v1-{{ checksum "go.mod" }}
+
+      - run: go mod download
```

Comment:
Just wanted to point out that I believe this is optional. Go will automatically fetch dependencies as needed.

Reply:
I am late to the party but in general, for caching to work, _a_ go command needs to run such that dependencies are fetched to populate the `save_cache` path(s). Since nothing runs before this, the `save_cache` will never have anything to save and therefore wouldn't ever cache anything. 

The way this works in practice is while `go mod download` runs every build, for any subsequent builds (after a cache with the same checksum was saved previously), it will just not do anything and go on to the next step as it has all it's dependencies restored from the cache.  

This is just an FYI--For this project we don't have any dependencies 🎉 so it doesn't matter 😄

Reply:
Since it doesn't matter for this one, I won't open a new PR, but thank you @alvin-huang! TIL!

Reply:
Ah! Good catch. Cache saving should occur towards the end of the job after go vet or test probably.